ReactiveCocoaTests/Swift/PropertySpec.swift

```diff
@@ -330,46 +334,803 @@ class PropertySpec: QuickSpec {
+
+						transformedProperty.signal.observeNext { _ in
+							counter += 1
+						}
+
+						property.value = initialPropertyValue
+						expect(counter) == 0
+
+						property.value = subsequentPropertyValue
+						expect(counter) == 1
+
+						property.value = finalPropertyValue
+						expect(counter) == 2
+
+						property.value = initialPropertyValue
+						expect(counter) == 2
+					}
+
+					it("should complete its producer only when the source property is deinitialized") {
+						var counter = 0
+						var completed = false
+
+						var transformedProperty = Optional(property.skipRepeats())
+						transformedProperty!.producer.start { event in
+							switch event {
+							case .Next:
+								counter += 1
+							case .Completed:
+								completed = true
+							default:
+								break
+							}
+						}
+
+						expect(counter) == 1
+
+						property.value = initialPropertyValue
+						expect(counter) == 1
+
+						transformedProperty = nil
+						expect(completed) == false
+
+						property = nil
+						expect(completed) == true
+					}
+				}
+
+				describe("uniqueValues") {
+					it("should emit hashable values that have not been emited before") {
+						let transformedProperty = property.uniqueValues()
+
+						var counter = 0
+						transformedProperty.signal.observeNext { _ in
+							counter += 1
+						}
+
+						property.value = initialPropertyValue
+						expect(counter) == 0
+
+						property.value = subsequentPropertyValue
+						property.value = subsequentPropertyValue
+
+						expect(counter) == 1
+
+						property.value = finalPropertyValue
+						property.value = initialPropertyValue
+						property.value = subsequentPropertyValue
+
+						expect(counter) == 2
+					}
+
+					it("should emit only the values of which the computed identity have not been captured before") {
+						let transformedProperty = property.uniqueValues { _ in 0 }
+
+						var counter = 0
+						transformedProperty.signal.observeNext { _ in
+							counter += 1
+						}
+
+						property.value = initialPropertyValue
+						property.value = subsequentPropertyValue
+						property.value = finalPropertyValue
+						expect(counter) == 0
+					}
+
+					it("should complete its producer only when the source property is deinitialized") {
+						var counter = 0
+						var completed = false
+
+						var transformedProperty = Optional(property.uniqueValues())
+						transformedProperty!.producer.start { event in
+							switch event {
+							case .Next:
+								counter += 1
+							case .Completed:
+								completed = true
+							default:
+								break
+							}
+						}
+
+						expect(counter) == 1
+
+						property.value = initialPropertyValue
+						expect(counter) == 1
+
+						transformedProperty = nil
+						expect(completed) == false
+
+						property = nil
+						expect(completed) == true
+					}
+				}
+			}
+
+			describe("flattening") {
+				describe("flatten") {
+					describe("FlattenStrategy.Concat") {
```

Comment:
These 3 levels seem unnecessary, I think you can reduce it to just `flattening` and `FlattenStrategy.Concat`.